hack/lib.bash

```
@@ -0,0 +1,364 @@
+        kubectl get secret admin-kubeconfig -o jsonpath='{.data.kubeconfig}' \
+        | base64 -d \
+        > "$kcp_kubeconfig" \
+        || die "Failed to get admin kubeconfig from kind cluster"
+
+    # Replace the port with the node port from the service
+    yq -i ".clusters[].cluster.server |= sub(\":6443\"; \":8443\")" "$kcp_kubeconfig"
+
+    # Create port forward to access kcp from host
+    kcp::front_proxy_forward "$kind_kubeconfig" "8443"
+    cp "$kcp_kubeconfig" "$kcp_host_kubeconfig"
+    local hostname="$(kubectl --kubeconfig "$kind_kubeconfig" get rootshards.operator.kcp.io root -o jsonpath='{.spec.external.hostname}')"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)